sklearn/pipeline.py

```diff
@@ -1615,7 +1642,20 @@ def fit_transform(self, X, y=None, **fit_params):
             The `hstack` of results of transformers. `sum_n_components` is the
             sum of `n_components` (output dimension) over transformers.
         """
-        results = self._parallel_func(X, y, fit_params, _fit_transform_one)
+        if _routing_enabled():
+            routed_fit_transform_params = process_routing(
+                self, "fit_transform", **fit_params
+            )
+        else:
+            # TODO(SLEP6): remove when metadata routing cannot be disabled.
+            routed_fit_transform_params = Bunch()
+            for name, _ in self.transformer_list:
+                routed_fit_transform_params[name] = Bunch(fit_transform={})
+                routed_fit_transform_params[name].fit_transform = fit_params
```

Comment:
```suggestion
                routed_fit_transform_params[name].fit_transform = fit_params
                routed_fit_transform_params[name].fit = fit
```

Reply:
I believe it needs to be like that:

```python
                if hasattr(obj, "fit_transform"):
                    routed_params[name] = Bunch(fit_transform={})
                    routed_params[name].fit_transform = params
                else:
                    routed_params[name] = Bunch(fit={})
                    routed_params[name] = Bunch(transform={})
                    routed_params[name].fit = params
                    routed_params[name].transform = params
```
It accounts for the metadata to be passed to the correct methods of the sub-transformer. I've also added a test for that (`test_feature_union_fit_params_without_fit_transform`).